Keithley2612B_voltage_sweep.py

```diff
@@ -164,36 +235,41 @@ def sweep_operation(smu_id, steps_no, measure_delay, nplc, start, end):
 
             target_volt = get_target_volt(start_volt)
             
-            steps_num = int (input ('Enter the number of steps'))
+            steps_num = int (input ('Enter the number of steps: '))
 
             integration_time = get_integration_time()
             
-            measure_delay = float (input('Set settling delay (in seconds) before each measurement: (NOTE: Setting the value to -1 automatically starts taking measurement as soon as current is stable) '))
+            measure_delay = 0
```

Comment:
Wouldn't it be better to set this value to -1 so that the instrument waits for a stable value before taking measurement.

Reply:
Agree in principle, ideally we don't want fluctuations in the voltage. 

I think however, waiting for a stable voltage might affect the scan rate calculations. I am not sure if the Keithley does its own subtraction. For example, if we set a source delay of 5s (an extreme value), and it takes the Keithley 2s to stabilise its voltage; does the Keithley automatically reduce the source delay down to 3s, so that the total delay time is still just 5s? 

Total delay time = Source delay + measurement delay + Integration Time, and the value we are really concerned with is the total delay time, because that will be related to the scan rate. I am not sure if it works this way, but it seems setting measurement delay to -1 means that Keithley will decide for itself the value of the measurement delay. That makes it hard to control what the source delay should be, or if it is needed at all. 

The alternative I have is to define the measurement delay ourselves. That way, only the source delay needs to be changed. But then there is no guarantee that the measurement delay we chose would ensure a stable voltage.

Reply:
Yeah I understand that. Maybe we don't have to worry about that if the settled value and the value at 0 delay is not a lot different. We can probably try the same tests with -1 delay to confirm this.